x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/TransportRollupAction.java

```diff
@@ -144,8 +149,24 @@ protected void masterOperation(
         ActionListener<AcknowledgedResponse> listener
     ) {
         String sourceIndexName = request.getSourceIndex();
-        IndexMetadata sourceIndexMetadata = state.getMetadata().index(sourceIndexName);
+
+        final IndicesAccessControl indicesAccessControl = threadContext.getTransient(AuthorizationServiceField.INDICES_PERMISSIONS_KEY);
+        if (indicesAccessControl != null) {
+            final IndicesAccessControl.IndexAccessControl indexPermissions = indicesAccessControl.getIndexPermissions(sourceIndexName);
+            if (indexPermissions != null) {
+                boolean hasDocumentLevelPermissions = indexPermissions.getDocumentPermissions().hasDocumentLevelPermissions();
+                boolean hasFieldLevelSecurity = indexPermissions.getFieldPermissions().hasFieldLevelSecurity();
+                if (hasDocumentLevelPermissions || hasFieldLevelSecurity) {
```

Comment:
I thought we had agreed that if security is enabled on the index, downsampling would be entirely not allowed. 

Here, I see it is disabled only if DLS or FLS is enabled. This may make sense, but I wanted to make sure we are all onboard with this decision 

cc @nik9000 @martijnvg

Reply:
I discussed this with @martijnvg , there are two levels of security here:
1. security associated to the user/role that hits the API endpoint: this is the first level of security. `TransportRollupAction` uses "indices:admin/xpack/downsample" which means that calling downsample API requires `admin` privileges
2. if the API call is allowed then we check if DLS or FLS is defined.

This code is checking just (2). Responsibility of (1) does not belong to "application logic" but is a cross cutting concern that is delegated to security interceptors.

See test named "Downsample user missing admin permissions to run the downsample operation" which tests (1) and the other 3 tests which check the other 3 cases:
1. Admin with DLS => fails because of DLS
2. Admin with FLS => fials because of FLS
3. Admin without DLS and without FLS => succeeds because it has admin rights and no DLS or FLS defined

Reply:
@csoulios is there any reason why this was implemented requiring admin permissions to run the downsample operation? Should't be better to have a specific set of permissions (something like `downsample` role) which allows running downsampling if the user has:
1. read source index permissions
2. create and write target index permissions
3. any other permission required to do things like copy settings/mappings/...

Reply:
It was first implemented with the strictest permissions. What you write sounds reasonable to me